src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -362,8 +418,34 @@ public Task<Result> InvokeWithCancellationAsync<Result>(string targetName, IRead
         /// <exception cref="ObjectDisposedException">If this instance of <see cref="JsonRpc"/> has been disposed.</exception>
         public Task NotifyAsync(string targetName, params object[] arguments)
         {
+            // If somebody calls InvokeInternal<T>(id, "method", null), the null is not passed as an item in the array.
+            // Instead, the compiler thinks that the null is the array itself and it'll pass null directly.
+            // To account for this case, we check for null below.
+            arguments = arguments ?? new object[] { null };
```

Comment:
So this line was in InvokeAsync to begin with.  I only added it after adding my unit tests and one of them failed, then I noticed that this line was only present in InvokeAsync and not NotifyAsync.  Talked to Ilya and we thought it was just because this line was accidentally left out of the NotifyAsync?  If we want to remove it, it should be done in both places.